#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From a98e7f8d9e46845847f434115d800249f08627b8 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Fri, 14 Aug 2020 12:30:41 +0200
Subject: [PATCH] w64: Use alternate regview instead of ifdefed keys

* src/common.cpp (_read_regStr): Use alterative view
for HKCU, too.
* src/common_indep.h (GPG4WIN_INSTDIR): Remove ifdefed names.

--
This fixes per user install of GpgOL and makes it easier
to avoid problems with different regviews if we always check
both.
---
 src/common.cpp     |  5 ++++-
 src/common_indep.h | 12 ++----------
 2 files changed, 6 insertions(+), 11 deletions(-)

diff --git a/src/common.cpp b/src/common.cpp
index a4e9c7d..2b7857d 100644
--- a/src/common.cpp
+++ b/src/common.cpp
@@ -187,7 +187,10 @@ readRegStr (const char *root, const char *dir, const char *name)
         return ret;
     }
     ret = _readRegStr (root_key, dir, name, false);
-
+    if (ret.empty()) {
+        // Try alternate as fallback
+        ret = _readRegStr (root_key, dir, name, true);
+    }
     if (ret.empty()) {
         // Try local machine as fallback.
         ret = _readRegStr (HKEY_LOCAL_MACHINE, dir, name, false);
diff --git a/src/common_indep.h b/src/common_indep.h
index 4c43b10..aad03a8 100644
--- a/src/common_indep.h
+++ b/src/common_indep.h
@@ -40,16 +40,8 @@
 #endif

 /* The Registry key used by Gpg4win.  */
-#ifdef _WIN64
-# define GPG4WIN_REGKEY_2  "Software\\Wow6432Node\\GNU\\GnuPG"
-#else
-# define GPG4WIN_REGKEY_2  "Software\\GNU\\GnuPG"
-#endif
-#ifdef _WIN64
-# define GPG4WIN_REGKEY_3  "Software\\Wow6432Node\\Gpg4win"
-#else
-# define GPG4WIN_REGKEY_3  "Software\\Gpg4win"
-#endif
+#define GPG4WIN_REGKEY_2  "Software\\GNU\\GnuPG"
+#define GPG4WIN_REGKEY_3  "Software\\Gpg4win"
 /* Identifiers for the protocol.  We use different one than those use
    by gpgme.  FIXME: We might want to define an unknown protocol to
    non-null and define such a value also in gpgme. */
--
2.20.1
